### DISCUSSION TITLE
How to get originalName when modifying the collection name

### DISCUSSION THREAD

**Discussion by lnlion** [pocketbase/pocketbase#1526] [Q&A] [answered]
How to get originalName in the OnCollectionBeforeUpdateRequest or OnCollectionAfterUpdateRequest event when modifying the collection name？

**Comment by ganigeorgiev** [comment, marked as answer]
You should be able to fetch the original/old collection by querying the non-transaction `Dao()` in the before hook, something like (_I haven't tested it_):

```go
package main

import (
	"log"

	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/core"
)

func main() {
	app := pocketbase.New()

	app.OnCollectionBeforeUpdateRequest().Add(func(e *core.CollectionUpdateEvent) error {
		old, err := app.Dao().FindCollectionByNameOrId(e.Collection.Id)
		if err != nil {
			return err
		}

		log.Println(old.Name, e.Collection.Name)

		return nil
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}
```

**Reply by lnlion** [reply]
Thanks, Problem solved；Your response to questions is really fast. hope this project can get better and better
